include/deal.II/differentiation/ad/ad_drivers.h

```diff
@@ -622,39 +786,176 @@ namespace Differentiation
     {
       using scalar_type = double;
 
+      TapedDrivers()
+        : active_tape(numbers::invalid_tape_index)
+        , keep_values(true)
+        , is_recording_flag(false)
+        , use_stored_taped_buffer_sizes(false)
+        , obufsize(0u)
+        , lbufsize(0u)
+        , vbufsize(0u)
+        , tbufsize(0u)
+      {}
+
       // === Taping ===
 
-      static void
-      start_taping(const types::tape_index tape_index,
-                   const bool              keep_independent_values)
+      bool
+      is_recording() const
       {
-        trace_on(tape_index, keep_independent_values);
+        return is_recording_flag;
       }
 
-      static void
-      start_taping(const types::tape_index        tape_index,
-                   const bool                     keep_independent_values,
-                   const types::tape_buffer_sizes obufsize,
-                   const types::tape_buffer_sizes lbufsize,
-                   const types::tape_buffer_sizes vbufsize,
-                   const types::tape_buffer_sizes tbufsize)
+      types::tape_index
+      active_tape_index() const
       {
-        trace_on(tape_index,
-                 keep_independent_values,
-                 obufsize,
-                 lbufsize,
-                 vbufsize,
-                 tbufsize);
+        return active_tape;
       }
 
-      static void
+      bool
+      keep_independent_values() const
+      {
+        return keep_values;
+      }
+
+      bool
+      is_registered_tape(const types::tape_index tape_index) const
+      {
+        // Sigh... This is a mess :-/
+        // The most succinct way to get this piece of information, would be to
+        // use the getTapeInfos() function, but would come at the expense of
+        // creating an inactive tape data within ADOL-C's global store. For
```

Comment:
Have you thought about asking the ADOL-C developers to provide this kind of functionality for you? Or could one easily write a patch for it? Since we're already requiring the use of a dev version of ADOL-C, maybe it would be possible for them to get the kind of function you need in before their next release.

Reply:
Maybe... I'll open up an issue on their GitLab repo to ask.